### PR TITLE
Implemented analogue camera controls in battle mode

### PR DESCRIPTION
**Pull request by CosmosXIII**
I added analogue controls for the battle camera. You can rotate the camera with the controller right stick and zoom-in/ zoom-out with the analogue triggers. Controls are only enabled when no camera animation is being executed.

### CODE REVIEW

**Review by julianxhokaxhiu — APPROVED** [review]
LGTM

**Comment by julianxhokaxhiu** [comment]
Thank you!

File: src/ff7/camera.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        struct point3d rotationOffset = { 0.0, 0.0, 0.0 };
+        float zoomSpeed = 0.0f;
+        float zoomOffset = 0.0f;
+        const float minZoomDist = 5000.0f;
+        const float maxZoomDist = 30000.0f;
+        const float minVerticalAngle = 5.0f;
+        const float maxVerticalAngle = 85.0f;
+};
+
+extern Camera camera;
```
Comment (sithlord48) [inline src/ff7/camera.h:42]: Newline should be at the EOF

File: src/ff7/camera.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/****************************************************************************/
+//    Copyright (C) 2021 Cosmos                                             //
```
Comment (sithlord48) [inline src/ff7/camera.h:2]: Its 2022 :d and you should use a real name anyone can claim to be "Cosmos"

File: src/ff7/misc.cpp [hunk 3]
```diff
@@ -248,20 +286,24 @@ struct ff7_gamepad_status* ff7_update_gamepad_status()
 			ff7_externals.gamepad_status->button5 = gamepad.IsPressed(XINPUT_GAMEPAD_LEFT_SHOULDER); // L1
 			ff7_externals.gamepad_status->button6 = gamepad.IsPressed(XINPUT_GAMEPAD_RIGHT_SHOULDER); // R1
 			ff7_externals.gamepad_status->button7 = gamepad.leftTrigger > 0.85f; // L2
-			ff7_externals.gamepad_status->button8 = gamepad.rightTrigger > 0.85f; // R2
+
+			// Remap R2 trigger to R3 so that we can use the trigger for zoom-in when analogue controls are enabled
```
Comment (sithlord48) [inline src/ff7/misc.cpp:290]: Please do not change controls if a feature is active. 
What happens if this is enabled and i want to remap w/ R2 / L2 do in battle ? 

I suggest
  Either use L3 / R3 for zoom 
                 or 
  Hold R2 or L2 and use the right stick's up / down  for zoom

**Review by sithlord48 — COMMENTED** [review]
Awesome Feature !

File: src/ff7/camera.h (hunk 2 above)
Comment (julianxhokaxhiu) [reply src/ff7/camera.h:2]: I updated it overall. Good catch, thanks!

File: src/ff7/camera.h (hunk 1 above)
Comment (julianxhokaxhiu) [reply src/ff7/camera.h:42]: Not really mandatory, but it's a suggested Unix best practice.

**Comment by Tharsos83** [comment]
Hi, I noticed the following about this feature: The speedhack is activated via controller with L3 + L2/R2 (Playstation). L2/R2 buttons are also used for zooming in fights. This means that you always zoom in during battle when you switch on the speedhack.

**Pull request merged**